### DISCUSSION TITLE
Set default value for conversion from json to integer

### DISCUSSION THREAD

**Discussion by jjcasmar** [nlohmann/json#2773] [Q&A] [answered]
In Qt, when you want to convert a QJSonObject to integer, you can do `j.toInt(-1);`, which returns the integer stored in the json value or -1 if the json value is not an integer. 

In nlohmann::json, to achieve the same, you need manually check if the value is an integer
```
int i = j.is_number_integer() ? j.get<int>() : -1;
```
This is of course more code. Is there any way to specify the default value for `.get()`?

**Comment by nlohmann** [comment, marked as answer]
No, there is no function for this.
